src/Sylius/Bundle/ArchetypeBundle/spec/EventListener/ArchetypeUpdateListenerSpec.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace spec\Sylius\Bundle\ArchetypeBundle\EventListener;
+
+use Doctrine\Common\Persistence\ObjectManager;
+use Doctrine\Common\Persistence\ObjectRepository;
+use PhpSpec\ObjectBehavior;
+use Prophecy\Argument;
+use Sylius\Bundle\ArchetypeBundle\Exception\IncompatibleCollaboratorException;
+use Sylius\Component\Archetype\Builder\ArchetypeBuilderInterface;
+use Sylius\Component\Archetype\Model\ArchetypeInterface;
+use Sylius\Component\Archetype\Model\ArchetypeSubjectInterface;
+use Sylius\Component\Resource\Exception\UnexpectedTypeException;
+use Symfony\Component\EventDispatcher\GenericEvent;
+
+/**
+ * @author Adam Elsodaney <adam.elso@gmail.com>
+ */
+class ArchetypeUpdateListenerSpec extends ObjectBehavior
+{
+    function let(ArchetypeBuilderInterface $builder, ObjectRepository $subjectRepository, ObjectManager $subjectManager)
+    {
+        $subjectRepository->getClassName()->willReturn(ArchetypeSubjectInterface::class);
+
+        $this->beConstructedWith($builder, $subjectRepository, $subjectManager);
+    }
+
+    function it_is_initializable()
+    {
+        $this->shouldHaveType('Sylius\Bundle\ArchetypeBundle\EventListener\ArchetypeUpdateListener');
+    }
+
+    function it_is_not_initializable_if_the_collaborating_repository_is_not_for_archetype_subjects(ArchetypeBuilderInterface $builder, ObjectRepository $nonSubjectRepository, ObjectManager $subjectManager)
+    {
+        $nonSubjectRepository->getClassName()->willReturn(NotAnArchetypeSubject::class);
```

Comment:
I think returning stdClass will be totally enough. There is no need of creating new class.